apps/cowswap-frontend/src/modules/fiatAmount/hooks/useHigherUSDValue.ts

```diff
@@ -0,0 +1,135 @@
+import { useEffect, useMemo, useRef, useState } from 'react'
+
+import { Currency, CurrencyAmount, Price, Token } from '@uniswap/sdk-core'
+
+import { unstable_batchedUpdates as batchedUpdate } from 'react-dom'
+import { Nullish } from 'types'
+
+import { USDC } from 'legacy/constants/tokens'
+
+import { useIsWrapOrUnwrap } from 'modules/trade/hooks/useIsWrapOrUnwrap'
+import { useWalletInfo } from 'modules/wallet'
+
+import { useGetCoingeckoUsdPrice } from 'api/coingecko'
+import { useSafeMemo } from 'common/hooks/useSafeMemo'
+import useBlockNumber from 'lib/hooks/useBlockNumber'
+import tryParseCurrencyAmount from 'lib/utils/tryParseCurrencyAmount'
+
+interface GetPriceQuoteParams {
+  currencyAmount: Nullish<CurrencyAmount<Currency>>
+  error: Error | null
+  price: Price<Token, Currency> | null
+  isLoading: boolean
+}
+
+// common logic for returning price quotes
+function useGetPriceQuote({ price, error, currencyAmount, isLoading }: GetPriceQuoteParams): {
+  value: CurrencyAmount<Token> | null
+  isLoading: boolean
+} {
+  return useMemo(() => {
+    if (!price || error || !currencyAmount) return { value: null, isLoading }
+
+    try {
+      return { value: price.invert().quote(currencyAmount), isLoading }
+    } catch (error: any) {
+      return { value: null, isLoading }
```

Comment:
Why not return the error?

Reply:
I actually didn't change this code.
Sorry for the noise.
This file is a moved [useStablecoinPrice/index.ts](https://github.com/cowprotocol/cowswap/pull/3074/files#diff-931a80cf03895977369e7dba4f68ee9d31844632a1e13c8f57a0c0bd50f1ab4e)

Reply:
Also, this file is deleted in the next PR